src/NUnitFramework/framework/Internal/ThreadUtility.cs

```diff
@@ -68,6 +131,96 @@ public static void Kill(Thread thread, object stateInfo)
             if ( (thread.ThreadState & ThreadState.WaitSleepJoin) != 0 )
                 thread.Interrupt();
         }
+
+        /// <summary>
+        /// Schedule a threadpool thread to check on the aborting thread in case it's in a message pump native blocking wait
+        /// </summary>
+        private static void DislodgeThreadInNativeMessageWait(Thread thread, int nativeId)
+        {
+            if (nativeId == 0) throw new ArgumentOutOfRangeException(nameof(nativeId), "Native thread ID must not be zero.");
+
+            // Schedule a threadpool thread to check on the aborting thread in case it's in a message pump native blocking wait
+            Delay(ThreadAbortedCheckDelay, CheckOnAbortingThread, new CheckOnAbortingThreadState(thread, nativeId));
+        }
+
+        private static void CheckOnAbortingThread(object state)
+        {
+            var context = (CheckOnAbortingThreadState)state;
+
+            switch (context.Thread.ThreadState)
+            {
+                case ThreadState.Aborted:
+                    return;
+                case ThreadState.AbortRequested:
+                    PostThreadCloseMessage(context.NativeId);
+                    break;
+            }
+
+            Delay(ThreadAbortedCheckDelay, CheckOnAbortingThread, state);
+        }
+
+        private sealed class CheckOnAbortingThreadState
+        {
+            public Thread Thread { get; }
+            public int NativeId { get; }
+
+            public CheckOnAbortingThreadState(Thread thread, int nativeId)
+            {
+                Thread = thread;
+                NativeId = nativeId;
+            }
+        }
+
+
+
```

Comment:
For future reference, excess blank lines triggers my OCD like trailing whitespace does for you 😄